Assets/Scripts/InventoryAndItemsScripts/Items/Interactables/PlayerInteractor.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using UnityEngine.InputSystem;
+using static UnityEngine.InputSystem.InputAction;
+
+namespace Items.Interactables
+{
+    /// <summary>
+    /// Handles player interaction with interactable objects
+    /// </summary>
+    public class PlayerInteractor : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("The keybinds that control character interaction with NPCs and objects in the environment.")]
+        InputAction InteractAction;
+
+        /// <summary>
+        /// null if none are in range; 
+        /// otherwise, references the first interactable to come in range 
+        /// (and returns back to null once out of range again)
+        /// </summary>
+        private IInteractable interactableCurrentlyInRange = null;
+
+        private bool mayInteractWithCurrentInteractable = false;
+
+        private void OnEnable()
+        {
+            InteractAction.performed += Interact;
+            InteractAction.Enable();
+        }
+
+        private void OnDisable()
+        {
+            InteractAction.performed -= Interact;
+            InteractAction.Disable();
+        }
+
+        /// <summary>
+        /// Triggered when a PlayerInteractor that was previously not colliding
+        /// with an InteractionCollider begins colliding with one
+        /// (takes in a bool that is true when the criteria to interact has been met)
+        /// </summary>
+        public event Action<bool> OnInInteractionRadius;
+
+        public event Action OnLeaveInteractionRadius;
+
+        public event Action<bool> CanInteractUpdate;
```

Comment:
Are these for later use? As of now I don't see where they are ever assigned/called.

Reply:
they were used a bit ago when I had a completely different test script calling them. At the moment I'm keeping them later in case they may still be used so I don't need to re-code them, but if it's an issue I can remove them.

Reply:
I think good to keep for now!